cmd/restic/cmd_backup.go

```diff
@@ -710,7 +713,21 @@ func runBackup(opts BackupOptions, gopts GlobalOptions, term *termstatus.Termina
 	if !gopts.JSON && !opts.DryRun {
 		progressPrinter.P("snapshot %s saved\n", id.Str())
 	}
-	if !success {
+	if opts.SummaryFilename != "" {
+		sf, err := os.OpenFile(opts.SummaryFilename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
+		if err != nil {
+			return errors.Errorf("%s: appending to summary failed: %v", opts.SummaryFilename, err)
+		}
+		buf := new(bytes.Buffer)
+		if err := json.NewEncoder(buf).Encode(progressReporter.FinishSummary(id)); err != nil {
+			return errors.Errorf("encoding summary failed: %v", err)
+		}
+		fmt.Fprintf(sf, "%s", buf.String())
```

Comment:
Please use `sf.Write(buf.Bytes())` instead. There's no need for format string interpolation and multiple String <-> []byte conversions.